Drivers/Src/CANbus.c

```diff
@@ -104,6 +118,52 @@ static ErrorStatus CANbus_SendMsg(CANId_t id, CANPayload_t payload) {
 	return retVal;
 }
 
+static ErrorStatus CANbus_SendMsg_FaultState(CANId_t id, CANPayload_t payload) {
+	uint8_t txdata[8];
+	uint8_t data_length = 0;
+
+	// TODO: is it really best to keep the list of
+	//		 valid messages to be sending in the driver?
+	switch (id) {
+		// Handle messages with one byte of data
+		case TRIP:
+		case ALL_CLEAR:
+		case CONTACTOR_STATE:
+		case WDOG_TRIGGERED:
+		case CAN_ERROR:
+		case CHARGE_ENABLE:
+			data_length = 1;
+			memcpy(txdata, &payload.data.b, sizeof(payload.data.b));
+			break;
+
+		// Handle messages with 4 byte data
+		case CURRENT_DATA:
+		case SOC_DATA:
+			data_length = 4;
+			memcpy(txdata, &payload.data.w, sizeof(payload.data.w));
+			break;
+
+		// Handle messages with idx + 4 byte data
+		case VOLT_DATA:
+		case TEMP_DATA:
+			data_length = 5;
+			txdata[0] = payload.idx;
+			memcpy(&txdata[1], &payload.data.w, sizeof(payload.data.w));
+			break;
+
+		// Handle invalid messages
+		default:
+			return ERROR;	// Do nothing if invalid
+	}
+
+	ErrorStatus retVal;
+	do{
+		retVal = BSP_CAN_Write(id, txdata, data_length);
+	} while(retVal == ERROR);
```

Comment:
Lol, this doesn't have to be changed, but this could also be written as `while(BSP_CAN_Write(id, txdata, data_length) == ERROR)` since it would have to execute once. And we wouldn't need the extra variable.